File: LeetCode/1598. Crawler Log Folder/1598.cpp
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int depth = 0;
        for (string log : logs) {
            if (log == "../") {
                if (depth > 0) {
                    depth--;
                }
            } else if (log != "./") {
                depth++;
            }
        }
        return depth;
        
    }
};

//with stack
/*
#include <vector>
#include <string>
#include <stack>

class Solution {
public:
    int minOperations(std::vector<std::string>& logs) {
        std::stack<std::string> folderStack;
        for (const std::string& log : logs) {
            if (log == "../") {
                if (!folderStack.empty()) {
                    folderStack.pop();
                }
            } else if (log != "./") {
                folderStack.push(log);
            }
        }
        return folderStack.size();
    }
};
*/

//with queue
/*
#include <vector>
#include <string>
#include <queue>

class Solution {
public:
    int minOperations(std::vector<std::string>& logs) {
        std::queue<std::string> folderQueue;
        int depth = 0;
        for (const std::string& log : logs) {
            if (log == "../") {
                if (depth > 0) {
                    depth--;
                    
                }
            } else if (log != "./") {
                depth++;
                folderQueue.push(log);
            }
        }
        return depth;
    }
};
*/

// other people's solution

/*
class Solution {
public:
    int minOperations(vector<string>& logs) {
        int n = 0;
        for (int i = 0; i < logs.size(); i++) {
            if (logs[i] == "../") {
               if(n>0) n--;
            } else if (logs[i] == "./") {
                continue;
            } else {
                n++;
            }
        }
        if(n<0) return 0;
        return n;
    }
};
*/
/*
class Solution {
public:
    int minOperations(vector<string>& logs) {
        stack<string> s;
        for(string i:logs){
            if(i=="./") continue;
            else if(i=="../"){
                if(s.size()!=0) s.pop();
            }
            else{
                s.push(i);
            }
        }
        return s.size();
    }
};
*/

File: LeetCode/1598. Crawler Log Folder/1598.cs
public class Solution
{
    public int MinOperations(string[] logs)
    {
        //Initialize depth to 0
        int depth = 0;
        //Iterate through the logs
        foreach (var log in logs)
        {
            //If the log is "../" and depth is greater than 0, decrement the depth
            if (log == "../")
            {
                //If the depth is greater than 0, decrement the depth
                if (depth > 0)
                {
                    depth--;
                }
            }
            //If the log is "./", continue
            else if (log != "./")
            {
                depth++;
            }
        }
        return depth;

    }
}

//other people's solution
/*
public class Solution {
    public readonly string stay = @"./";
    public readonly string back = @"../";
    public int MinOperations(string[] logs) {
        int distance = 0;
        foreach(var log in logs){
            if(log[1] == '.' & distance != 0){
                distance--;
            }
            else if(log[0] != '.'){
                distance++;
            }
        }
        return distance;
    }
}
*/
/*
public class Solution {
    public int MinOperations(string[] logs) {
         int count = 0;
        foreach (string str in logs)
        {
            if (str == "../")
            {
                count--;
            }
            else if (str != "./")
            {
                count++;
            }
            if (count < 0)
            {
                count = 0;
            }
        }
        return count;
    }
}
*/
/*
public class Solution {
    public int MinOperations(string[] logs) {
        int count = 0;
        for(int i=0;i<logs.Length;i++){
            string optn = logs[i];
            if(optn.Equals("../"))
            {
                if(!(count == 0))
                    count--;

            }
            else if(!optn.Equals("./")){
                count++;
            }
           // Console.WriteLine("optn: "+optn+" count: "+count);
        }
        return count;
        
    }
}
*/
/*
public class Solution {
    public int MinOperations(string[] logs) {
        var stack = new Stack<string>();

        foreach(var log in logs)
        {
            if(log == "../")
            {
                if(stack.Count() == 0)
                {
                    continue;
                }
                else
                {
                    stack.Pop();
                }
            }
            else if(log == "./")
            {
                continue;
            }
            else
            {
                stack.Push(log);
            }
        }

        return stack.Count();
    }
}
*/